sync/class.jetpack-sync-actions.php

```diff
@@ -137,6 +146,34 @@ static function do_full_sync( $modules = null ) {
 		self::do_send_pending_data(); // try to send at least some of the data
 	}
 
+	static function minute_cron_schedule( $schedules ) {
+		if( ! isset( $schedules["1min"] ) ) {
+			$schedules["1min"] = array(
+				'interval' => 60,
+				'display' => __( 'Every minute' ) );
+		}
+		return $schedules;
+	}
+
+	// try to send actions for up to a minute
+	static function do_cron_sync() {
+		if ( ! self::sync_allowed() ) {
+			return;
+		}
+
+		self::initialize_sender();
+
+		$start_time = microtime( true );
+
+		do {
+			self::$sender->do_sync();
+		} while ( 
+			self::$sender->get_sync_queue()->size() > 0 
```

Comment:
I think effectively this is the same as firing a new do_sync right here: https://github.com/Automattic/jetpack/pull/4471/files#diff-83cec40761e782fd7ccb955fb7c1d19aL161

this `while` will call `do_sync` too many times while `do_sync` is sending something